src/SFML/Graphics/Image.cpp

```diff
@@ -99,21 +101,30 @@ void Image::create(unsigned int width, unsigned int height, const Uint8* pixels)
 {
     if (pixels && width && height)
     {
+        // Create a new pixel buffer first for exception safety's sake
+        std::vector<Uint8> newPixels(width * height * 4);
+
+        // Copy the pixels
+        std::memcpy(&m_pixels[0], pixels, newPixels.size()); // faster than vector::assign
```

Comment:
Doing a separate `memcpy` after the vector is already initialized is more work than necessary. The `memcpy` can be done as part of the initialization thus saving the internal `memset`.